weblate/fonts/views.py

```diff
@@ -83,23 +86,67 @@ def post(self, request: AuthenticatedHttpRequest, **kwargs):
 @method_decorator(login_required, name="dispatch")
 class FontDetailView(ProjectViewMixin, DetailView):
     model = Font
+    _font_form = None
 
     def get_queryset(self):
         return self.project.font_set.all()
 
     def get_context_data(self, **kwargs):
         result = super().get_context_data(**kwargs)
         result["can_edit"] = self.request.user.has_perm("project.edit", self.project)
+        result["font_form"] = self._font_form or FontForm()
         return result
 
     def post(self, request: AuthenticatedHttpRequest, **kwargs):
         self.object = self.get_object()
+
         if not request.user.has_perm("project.edit", self.project):
             raise PermissionDenied
 
-        self.object.delete()
-        messages.error(request, gettext("Font deleted."))
-        return redirect("fonts", project=self.project.slug)
+        if "font" not in request.FILES:
+            messages.error(request, "No file uploaded.")
+            return self.get(request, **kwargs)
+
+        new_file = request.FILES["font"]
+
+        try:
+            uploaded_family, uploaded_style = get_font_name(new_file)
+            new_file.seek(0)
+        except Exception:
+            messages.error(
+                request,
+                "Could not extract font metadata. Ensure it’s a valid font file.",
+            )
+            return self.get(request, **kwargs)
+
+        # Enforce same family & style
+        if uploaded_family != self.object.family or uploaded_style != self.object.style:
+            messages.error(
+                request,
+                f"The uploaded font must match the existing family and style: "
+                f"'{self.object.family} {self.object.style}'",
+            )
+            return self.get(request, **kwargs)
+
+        # Compare file content hashes
+        new_hash = hashlib.sha256(new_file.read()).hexdigest()
+        new_file.seek(0)
+
+        current_hash = hashlib.sha256(self.object.font.read()).hexdigest()
+        self.object.font.seek(0)
+
+        if new_hash == current_hash:
+            messages.info(
+                request, "The uploaded font file is identical to the current one."
```

Comment:
Localize the message:

```suggestion
                request, gettext("The uploaded font file is identical to the current one.")
```